src/build.cc

```diff
@@ -460,12 +462,26 @@ void RealCommandRunner::Abort() {
   subprocs_.Clear();
 }
 
-bool RealCommandRunner::CanRunMore() const {
+size_t RealCommandRunner::CanRunMore() const {
   size_t subproc_number =
       subprocs_.running_.size() + subprocs_.finished_.size();
-  return (int)subproc_number < config_.parallelism
-    && ((subprocs_.running_.empty() || config_.max_load_average <= 0.0f)
-        || GetLoadAverage() < config_.max_load_average);
+
+  long capacity = config_.parallelism - subproc_number;
```

Comment:
No code should ever use `long`. It is evil.
It changes size based on the operating system used, and not the underlying processor architecture. This leads to lots of surprises.

Use either `size_t` or `ssize_t`

Reply:
The logic requires `capacity` to become negative, so `size_t` is not an option. And since `ssize_t` only guarantees to be able to hold one negative value, i.e.,  `-1`, it is also appears to be no viable option here.

Reply:
can you use int64_t instead?